airflow/migrations/versions/cc1e65623dc7_add_max_tries_column_to_task_instance.py

```diff
@@ -0,0 +1,75 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""add max tries column to task instance
+
+Revision ID: cc1e65623dc7
+Revises: 127d2bf2dfa7
+Create Date: 2017-06-19 16:53:12.851141
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'cc1e65623dc7'
+down_revision = '127d2bf2dfa7'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+from airflow import settings
+from airflow.models import DagBag, TaskInstance
+
+
+def upgrade():
+    op.add_column('task_instance', sa.Column('max_tries', sa.Integer,
+        server_default="-1"))
+    # Check if table task_instance exist before data migration. This check is
+    # needed for database like postgres that does not actually create table
+    # until migration finishes. Checking task_instance table exists prevent
+    # the error of querying non-existing task_instance table.
```

Comment:
There might be ways to make postgres work in this case but it make sense to not perform data migration if there is no task_instance table. I tested locally with postgres with task instance table and it's working.